### PR TITLE
[Feature] Accelerations Tab and Flyout Skeletons

### PR DESCRIPTION
**Pull request by sejli**
### Description
All added UI are currently skeletons with no backend logic linked.
* Change data sources tiles
* Change data sources header box
* Add accelerations tab in data sources page
* Add acceleration details flyout 
  * Add flyout tabs with details, schema, and SQL definition
* Rearrange `associated_objects` to separate directory

image
image
image
image
image

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    sort: {
+      field: 'name',
+      direction: 'asc',
+    },
+  };
+
+  return (
+    <>
+      <EuiSpacer />
+      {isFlyoutVisible && (
```
Comment (ps48) [inline public/components/datasources/components/manage/accelerations/acceleration_table.tsx:182]: Did we try to see if this flyout open's from other plugins? I would recommend to use overlay service as mentioned here: https://github.com/opensearch-project/OpenSearch-Dashboards/tree/790c076fd4c8f38ade5482cede6d92d10792be54/src/plugins/opensearch_dashboards_react#overlays-service

Comment (RyanL1997) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:182]: +1 im also looking into this.

File: public/components/datasources/components/manage/accelerations/flyout_modules/accelerations_schema_tab.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import React from 'react';
+
+interface AccelerationSchemaTabProps {
+  acceleration: any;
+}
+
+export const AccelerationSchemaTab = (props: AccelerationSchemaTabProps) => {
+  const { acceleration } = props;
+  // TODO: Use schema returned from backend
+  console.log(acceleration);
```
[the remark this answers is not in the document]
Comment (sejli) [reply public/components/datasources/components/manage/accelerations/flyout_modules/accelerations_schema_tab.tsx:16]: Added random `console.log()` lines to indicate where I need to add backend logic. I'll make them a bit more verbose.

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  const RefreshButton = () => {
+    // TODO: Implement logic for refreshing acceleration
+    return (
+      <>
+        <EuiButton onClick={() => console.log}>Refresh</EuiButton>
+      </>
+    );
+  };
+
+  const CreateButton = () => {
```
[the remark this answers is not in the document]
Comment (sejli) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:46]: I think React components should be written using `PascalCase` per the [OpenSearch-Dashboards linter config](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/opensearch-eslint-config-opensearch-dashboards/typescript.js#L151).

**Review by paulstn — APPROVED** [review]
Upon addressing the use for overlay service for flyout comment, everything would look good to me. Dropping an approval.

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (RyanL1997) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:46]: Oh.. seems like I need to change bunch of my naming in my PR 😄

[the remark this answers is not in the document]
Comment (RyanL1997) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:46]: then you can just ignore these comments

**Comment by kamingleung** [comment]
Can we move the taps into part of the header, checkout the OUI example under [Complicated flyout](https://oui.opensearch.org/1.5/#/layout/flyout#more-complicated-flyout)

Like this:
image

**Comment by sejli** [comment]
> Can we move the taps into part of the header, checkout the OUI example under [Complicated flyout](https://oui.opensearch.org/1.5/#/layout/flyout#more-complicated-flyout)

Sure, I can make that change. Here's a before and after.
## Before:
image

## After:
image
image
image

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx (hunk 1 above)
Comment (sejli) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:182]: Using the overlay service to render flyouts:
```
const renderAccelerationDetailsFlyout = (acceleration: any) =>
  core.overlays.openFlyout(
    toMountPoint(<AccelerationDetailsFlyout acceleration={acceleration} />)
  );
setRenderAccelerationDetailsFlyout(renderAccelerationDetailsFlyout);

return {
  renderAccelerationDetailsFlyout,
};
```

**Comment by kamingleung** [comment]
@sejli Thanks for the updates! From the After screenshots, looks like the tabs and tab contents have moved into the flyout header? I wonder is it possible to have the tab controls anchored within the header and have the tab contents display in the flyout body?

**Comment by sejli** [comment]
Hey @kamingleung, I played around with it for a bit, was able to use `OuiTabs` and `OuiTab` to render the tabs in the header and the tab content in the body. Are there any specific margins I should be using? 

Here is a screenshot **without** any style adjustments:
image

Here is a screenshot with a `-15px` margin on the bottom and top of the header and footer respectively:
image

File: public/plugin.tsx [hunk 4]
```diff
@@ -79,6 +80,9 @@ import {
   ObservabilityStart,
   SetupDependencies,
 } from './types';
+import { AccelerationDetailsFlyout } from './components/datasources/components/manage/accelerations/acceleration_details_flyout';
+import { createGetterSetter } from '../../../src/plugins/opensearch_dashboards_utils/public';
+import { toMountPoint } from '../../../src/plugins/opensearch_dashboards_react/public/';
```
Comment (ps48) [inline public/plugin.tsx:85]: Do we need to mention add `opensearchDashboardsReact` as a plugin dependency in `dashboards.json` file?

Comment (sejli) [reply public/plugin.tsx:85]: `opensearchDashboardsReact` is already a required plugin in `opensearch_dashboards.json`.

**Review by ps48 — APPROVED** [review]
LGTM, let's fix the conflicts.

**Comment by kamingleung** [comment]
> Here is a screenshot **without** any style adjustments:

@sejli Let's do **without** any style adjustments. Thanks.

**Pull request merged**